### PR TITLE
Bump sabre

### PR DESCRIPTION
**Pull request by rullzer**
- [x] Requires https://github.com/nextcloud/3rdparty/pull/20

Uses: https://github.com/owncloud/core/pull/26115

Lets see what CI thinks of this!

### CODE REVIEW

**Comment by phsc** [comment]
#1427

**Comment by MorrisJobke** [comment]
> Lets see what CI thinks of this!

It says 🚫 ❗️ 

```
1) OCA\DAV\Tests\unit\CalDAV\CalendarTest::testPrivateClassification with data set #0 (3, false)
Undefined index: {http://owncloud.org/ns}owner-principal
/drone/src/github.com/nextcloud/server/apps/dav/lib/CalDAV/Calendar.php:286
/drone/src/github.com/nextcloud/server/apps/dav/lib/CalDAV/Calendar.php:208
/drone/src/github.com/nextcloud/server/apps/dav/tests/unit/CalDAV/CalendarTest.php:230
2) OCA\DAV\Tests\unit\CalDAV\CalendarTest::testConfidentialClassification with data set #0 (3, false)
Undefined index: {http://owncloud.org/ns}owner-principal
/drone/src/github.com/nextcloud/server/apps/dav/lib/CalDAV/Calendar.php:286
/drone/src/github.com/nextcloud/server/apps/dav/lib/CalDAV/Calendar.php:193
/drone/src/github.com/nextcloud/server/apps/dav/tests/unit/CalDAV/CalendarTest.php:319
```

**Comment by rullzer** [comment]
Yeah I know. Have been messing with it to get the caldav intergration tests happy. But no real luck so far

**Comment by rullzer** [comment]
Whee CI is happy. Review time!

CC: @nickvergessen @MorrisJobke @LukasReschke @icewind1991

**Comment by MorrisJobke** [comment]
Works here fine 👍 I did some smoke testing with WebDAV and some file operations.

**Comment by MorrisJobke** [comment]
@nickvergessen could we get smashbox for this?

**Comment by nickvergessen** [comment]
Smashbox passed ✅

File: apps/dav/lib/CalDAV/Schedule/Plugin.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+class Plugin extends \Sabre\CalDAV\Schedule\Plugin {
+
+	/**
+	 * Returns a list of addresses that are associated with a principal.
+	 *
+	 * @param string $principal
+	 * @return array
+	 */
+	protected function getAddressesForPrincipal($principal) {
```
Comment (LukasReschke) [inline apps/dav/lib/CalDAV/Schedule/Plugin.php]: Missing tests. Will quickly add myself.

File: apps/dav/lib/CalDAV/Plugin.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+namespace OCA\DAV\CalDAV;
+
+use Sabre\HTTP\URLUtil;
+
+class Plugin extends \Sabre\CalDAV\Plugin {
+
+	/**
+	 * @inheritdoc
+	 */
+	function getCalendarHomeForPrincipal($principalUrl) {
```
Comment (LukasReschke) [inline apps/dav/lib/CalDAV/Plugin.php]: Missing tests. Will quickly add myself.

**Review by LukasReschke — CHANGES_REQUESTED** [review]
Looks good. Two new files are not really being tested. Will add myself quickly :)

**Review by LukasReschke — APPROVED** [review]

**Comment by LukasReschke** [comment]
Tests added and third-party rebased.

**Pull request merged**